crates/models/src/resources/deserialization.rs

```diff
@@ -0,0 +1,230 @@
+use bytes::Bytes;
+use schemars::JsonSchema;
+use serde::de::{MapAccess, Visitor};
+use serde::Deserialize;
+
+use crate::resources::ResourceDef;
+use crate::{ContentFormat, ContentType, Object};
+
+#[derive(Clone, Debug, Deserialize, JsonSchema)]
```

Comment:
This is impressive and looks like it was challenging to write... but as mentioned above I don't think we want to cojoin the Object vs Bytes representation of a ResourceDef at ser/deser time.

Instead use serde's existing support for untagged unions, and then we don't need to maintain custom Serializer / Visitor / Deser logic.